core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -501,6 +502,7 @@ object Parser extends ParserInstances {
 
   object Expectation {
     case class Str(offset: Int, str: String) extends Expectation
+    case class OneStr(offset: Int, strs: List[String]) extends Expectation
```

Comment:
I wonder if this should be a Set[String] or SortedSet[String] actually, since the order does not matter.

Also, I wonder if we really need Str and OneStr, maybe we just combine them into one error case. Having too many error results makes it harder to present simple messages to the users.